controller/RestDelivery.php

```diff
@@ -59,4 +67,95 @@ public function generate()
             $this->returnFailure($e);
         }
     }
+
+    /**
+     * Action to retrieve test compilation task status from queue
+     */
+    public function getStatus()
+    {
+        try {
+            if (!$this->hasRequestParameter(self::TASK_ID_PARAM)) {
+                throw new \common_exception_MissingParameter(self::TASK_ID_PARAM, $this->getRequestURI());
+            }
+            $data = $this->getTaskData($this->getRequestParameter(self::TASK_ID_PARAM));
+            $this->returnSuccess($data);
+        } catch (\Exception $e) {
+            $this->returnFailure($e);
+        }
+    }
+
+    /**
+     * @param $taskId
+     * @return array
+     */
+    protected function getTaskData($taskId)
+    {
+        $data = $this->traitGetTaskData($taskId);
+        $task = $this->getTask($taskId);
+        $report = \common_report_Report::jsonUnserialize($task->getReport());
+        $plainReport = $this->getPlainReport($report);
+        //the second report is report of compilation test
+        if (isset($plainReport[1]) && isset($plainReport[1]->getData()['uriResource'])) {
```

Comment:
I know that this is not the most elegant solution and not really reliable, but for now i did not found better solution to work with nested reports.